src/components/TransactionFlow/TxReceipt.tsx

```diff
@@ -471,38 +497,66 @@ export const TxReceiptUI = ({
           recipient={rawTransaction.to}
         />
       </div>
+      {txType === ITxType.FAUCET && (
+        <NewTabLink
+          href={`https://twitter.com/intent/tweet?text=${encodeURIComponent(
+            translateRaw('FAUCET_TWEET', {
+              $faucet_url: MYCRYPTO_FAUCET_LINK
+            })
+          )}`}
+        >
+          <Button inverted={true} fullwidth={true} className="TransactionReceipt-tweet">
+            <i className="sm-icon sm-logo-twitter TransactionReceipt-tweet-icon" />{' '}
+            <span className="TransactionReceipt-tweet-text">{translate('FAUCET_SHARE')}</span>
+          </Button>
+        </NewTabLink>
+      )}
       {completeButtonText && !(txStatus === ITxStatus.PENDING) && (
         <Button secondary={true} className="TransactionReceipt-another" onClick={resetFlow}>
           {completeButtonText}
         </Button>
       )}
-      {txStatus === ITxStatus.PENDING && txQueryType !== TxQueryTypes.SPEEDUP && txConfig && (
-        <Tooltip display="block" tooltip={translateRaw('SPEED_UP_TOOLTIP')}>
-          <Button
-            className="TransactionReceipt-another"
-            onClick={handleTxSpeedUpRedirect}
-            disabled={!supportsResubmit}
-          >
-            {translateRaw('SPEED_UP_TX_BTN')}
+      {txStatus === ITxStatus.PENDING &&
```

Comment:
Maybe add a prop flag to the conditional instead of using `txType` directly. We are trying to limit the amount of conditionals based on `txType` in the component to make it more readable.